### PR TITLE
fix ANR caused by DRM api

### PR DESCRIPTION
**Pull request by wenxi-zeng**
* fix DRM api ANR issue on certain device
* update coroutine to 1.6.0
* make it possible to inject coroutine settings for testing via inheritance

### CODE REVIEW

**Review by prayansh — APPROVED** [review]

File: android/src/main/java/com/segment/analytics/kotlin/android/plugins/AndroidContextPlugin.kt [hunk 1]
```diff
@@ -123,16 +125,43 @@ class AndroidContextPlugin : Plugin {
             emptyJsonObject
         }
 
+        // generate random identifier that does not persist across installations
+        // use it as the default device id in case DRM API failed to generate one.
+        val fallbackDeviceId = UUID.randomUUID().toString()
```
Comment (prayansh) [inline android/src/main/java/com/segment/analytics/kotlin/android/plugins/AndroidContextPlugin.kt:130]: is it better to have no device id or like a string of 0s to indicate not collected. and then use the fallback one when we fail to fetch via DRM?

File: android/src/main/java/com/segment/analytics/kotlin/android/plugins/AndroidContextPlugin.kt [hunk 2]
```diff
@@ -123,16 +125,43 @@ class AndroidContextPlugin : Plugin {
...
+
+        // run `getDeviceId` in coroutine, since the DRM API takes a long time
+        // to generate device id on certain devices and causes ANR issue.
+        analytics.analyticsScope.launch(analytics.analyticsDispatcher) {
+            var deviceId = fallbackDeviceId
+
+            // restrict getDeviceId to 2s to avoid ANR
+            // have to use a different scope than analyticsScope.
+            // otherwise, timeout cancellation won't work (i.e. the scope can't cancel itself)
+            val task = CoroutineScope(SupervisorJob()).async {
```
Comment (prayansh) [inline android/src/main/java/com/segment/analytics/kotlin/android/plugins/AndroidContextPlugin.kt:147]: why do we need to make another coroutineScope here? arent we already in one?

File: android/src/main/java/com/segment/analytics/kotlin/android/plugins/AndroidContextPlugin.kt (hunk 1 above)
Comment (wenxi-zeng) [reply android/src/main/java/com/segment/analytics/kotlin/android/plugins/AndroidContextPlugin.kt:130]: yeah. that's a good call out. my consideration is that in the case when DRM does fail, we have a consistent device id before and after (it takes 2s to know it's failed, could have some cached events). but using all 0s to indicate not ready to collect is also a good idea.

**Review by prayansh — APPROVED** [review]
i like the caching ❤️

**Pull request merged**